src/utils/TextDecoderPolyfill.js

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+"use strict";
+
+// Polyfill for TextDecoder.
+
+const REPLACEMENT_CHAR = '\uFFFD';
+
+export default class TextDecoder {
+    /**
+     * Decode a UTF-8 byte array as a javascript string
+     *
+     * @param {Uint8Array} u8Array UTF-8-encoded onput
+     * @return {str}
+     */
+    decode(u8Array) {
+        let u0, u1, u2, u3;
+
+        let str = '';
+        let idx = 0;
+        while (idx < u8Array.length) {
+            u0 = u8Array[idx++];
+            if (!(u0 & 0x80)) {
+                str += String.fromCharCode(u0);
```

Comment:
hrm. I'm not sure that's a valid assumption, though it appears to be the case: https://jsperf.com/vdh-string-concatenation.

Can you think of any other solutions?